src/compiler/factory.ts

```diff
@@ -139,7 +139,7 @@ namespace ts {
         return node;
     }
 
-    export function createTempVariable(recordTempVariable: (node: Identifier) => void, location?: TextRange): Identifier {
+    export function createTempVariable(recordTempVariable: ((node: Identifier) => void) | undefined, location?: TextRange): Identifier {

```

Comment:
To catch mistakes. Its acceptable to not record the temp variable, but you almost always want to. The only cases where we don't are when we're creating a temp parameter, or we are going to add the temp variable name to a `VariableDeclarationList` ourselves.